server/controllers/shared/risksAndNeedsController.test.ts

```diff
@@ -157,6 +167,53 @@ describe('RisksAndNeedsController', () => {
     })
   })
 
+  describe('roshAnalysis', () => {
+    it('renders the offence analysis page with the correct response locals', async () => {
+      const roshAnalysis = roshAnalysisFactory.build()
+      const previousBehaviourSummaryListRows = [{ key: { text: 'key-one' }, value: { text: 'value one' } }]
+
+      mockRoshAnalysisUtils.previousBehaviourSummaryListRows.mockReturnValue(previousBehaviourSummaryListRows)
+
+      when(oasysService.getRoshAnalysis).calledWith(username, person.prisonNumber).mockResolvedValue(roshAnalysis)
+
+      request.path = referPaths.show.risksAndNeeds.roshAnalysis({ referralId: referral.id })
+
+      const requestHandler = controller.roshAnalysis()
+      await requestHandler(request, response, next)
+
+      assertSharedDataServicesAreCalledWithExpectedArguments()
+
+      expect(response.render).toHaveBeenCalledWith('referrals/show/risksAndNeeds/roshAnalysis', {
+        ...sharedPageData,
+        hasRoshAnalysis: true,
+        importedFromText: `Imported from OASys on ${importedFromDate}.`,
```

Comment:
Not for this PR but I thought the same while doing the Lifestyle section I'm on at the moment, does it make sense to move this to `sharedPageData`?

Reply:
Given it shouldn't be in the `when the OASys service returns `null`` contexts, that might mean we can't use it those?